packages/marketplace/contracts/RoyaltiesRegistry.sol

```diff
@@ -110,23 +111,22 @@ contract RoyaltiesRegistry is OwnableUpgradeable, IRoyaltiesProvider {
     /// @notice returns royalties type from uint
     /// @param data in uint256
     /// @return royalty type
-    function _getRoyaltiesType(uint256 data) internal pure returns (uint256) {
-        for (uint256 i = 1; i <= ROYALTIES_TYPES_AMOUNT; ++i) {
+    function _getRoyaltiesType(uint256 data) internal pure returns (RoyaltyTypes) {
+        for (uint256 i = 1; i <= uint256(type(RoyaltyTypes).max); ++i) {
```

Comment:
uint256(type(RoyaltyTypes).max - 1 ?

Reply:
type().max ==  UNSUPPORTED_NONEXISTENT if you look at the old code  UNSUPPORTED_NONEXISTENT == ROYALTIES_TYPES_AMOUNT == 4. I don´t know why they used two names for the same thing.